### PR TITLE
added documentation for tracking scenarios api

### PR DESCRIPTION
**Pull request by patofp**
I will rebase from `main` branch and move to the new directory structure proposed in https://github.com/snowplow/documentation/pull/501 when it is merged. Is this acceptable @stanch?

Let me know if I missed something on the headers.  Thank you in advance 😅 

Updated changes:
* Update wording of Implementation Instructions everywhere (API) ✅
* Unify compatibility check sections ✅
* Reword “schema” for “data structure” ✅
* Update query params on endpoints ✅
* Check outstanding comments ✅
* Update the Nav according to Nick recommendations ^^

### CODE REVIEW

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+    style Deleted fill:#D63A31, stroke:#000000, stroke-width:1px;
+    linkStyle default stroke-width:2px,fill:#F2F4F7,stroke:#633EB5,color:#633EB5
+
+    Start(( )) -->|Create| Draft
+    Draft -->|Publish| Published
+    Draft -->|Delete| Deleted
+    Published -->|Deprecate| Deprecated
+    Deprecated -->|Undeprecate| Draft
+    Deleted((Deleted))
+```
```
Comment (gleb-lobov) [inline docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md:314]: 🔥

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 2]
```diff
@@ -0,0 +1,367 @@
+---
+title: "🆕 Managing tracking scenarios via the API"
```
Comment (stanch) [inline docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md:2]:
```suggestion
title: "Managing tracking scenarios via the API"
sidebar_label: "🆕 Managing tracking scenarios via the API"
```

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+                ]
+            }
+        },
+        "additionalProperties": false
+    }
+}
+...
+```
+
+This validation checks the compatibility of a particular tracking scenario version with the source Iglu schema version referenced by `event.source`.
```
Comment (stanch) [inline docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md:85]:
```suggestion
This validation checks the compatibility of a particular tracking scenario version with the source schema version referenced by `event.source`.
```

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+
+    Start(( )) -->|Create| Draft
+    Draft -->|Publish| Published
+    Draft -->|Delete| Deleted
+    Published -->|Deprecate| Deprecated
+    Deprecated -->|Undeprecate| Draft
+    Deleted((Deleted))
+```
+
+In addition to this lifecycle and in conjunction with versioning, we enforce that when a tracking scenario is **published**, the versions between two published versions are **squashed**. For example:
```
Comment (stanch) [inline docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md:316]: I think I would call this “discarded” rather than “squashed”...

**Review by stanch — CHANGES_REQUESTED** [review]

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+Start integrating the Tracking Scenarios API into your workflows and take control of your tracking scenarios in a seamless and automated manner. If you have any questions or need assistance, feel free to reach out to our support team. Happy tracking scenario management!
+
+## Getting Started
+For detailed information about each request, including query parameters and response formats, refer to the corresponding [API documentation](https://console.snowplowanalytics.com/api/msc/v1/docs).
+
+### Authorizing in the API Documentation
+
+To be able to post sample requests in the documentation you need to click the `Authorize` button at the top of the document and authorize with your token. The value for the token field in each individual requests is overwritten by this authorization.
+
+Each request will need to include your company's `organizationID` which is a UUID that can be retrieved from the URL immediately following the .com when visiting console:
```
Comment (cksnp) [inline docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md:18]: "immediately following the .com" --> "immediately following `.com`"

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+
+```mdx-code-block
+import Block14551 from "@site/docs/reusable/14551/_index.md"
+
+<Block14551/>
+```
+
+## Response Format
+
+The Tracking Scenarios API uses a specific response format for success cases (`2xx`) and for cases where critical and non-critical errors can occur, such as (`4xx`).
```
[the remark this answers is not in the document]
Comment (cksnp) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md:30]: I agree, I would add that.

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+        "event": {
+            "source": "iglu:com.example/ui_actions/jsonschema/1-0-0"
+        }
+    },
+    "message": "initial draft"
+}
+```
+
+- `message`: An optional field to provide a message.
+- `scenario`: The definition of the tracking scenario, which should have a defined structure. See [Validations](#validations).
```
Comment (cksnp) [inline docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md:165]: I don't understand the point about having "a defined structure", is this a reference to the schema referenced by `event.source`?

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+
+### Compatibility Check
+
+If a `scenario.event` is provided during creation, a compatibility check will be performed between the provided event and the one referenced in Iglu by `scenario.event.source`.
+
+This check is performed for the version indicated by the Iglu URI, referred to as the **current** version. Additionally, the compatibility check is also run for the **latest** version. The compatibility check results in three possible values: **Compatible**, **SchemaIncompatible**, or **SchemaUndecidable**. The different errors will be provided in the `errors` field of the [response](#response-format).
+
+- If **Compatible**, no errors or warnings will be attached to the `errors` response field.
+- If **SchemaIncompatible**, an error will be attached to the `errors` response field when the **current** version is incompatible, resulting in a **422 Unprocessable Entity** response without being able to create the tracking scenario. If the **latest** version is incompatible, a warning will be attached to the `errors` field, but a **201 Created** response will still be returned.
+- If **SchemaUndecidable**, a warning will be attached to the `errors` response field.
```
Comment (cksnp) [inline docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md:287]: I'm wondering if, at this point, a citation of the paper we based all this work on would be useful.

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+    style A color:#3F2874, fill:#F0EBF8, stroke:#3F2874, stroke-width:1px;
+    style B color:#3F2874, fill:#F0EBF8, stroke:#3F2874, stroke-width:1px;
+    linkStyle default stroke-width:2px,fill:#F2F4F7,stroke:#633EB5,color:#633EB5
+
+    A[Published 1] --> B[Published 5]
+```
+
+### Deprecating a Tracking Scenario
+
+Tracking scenarios can also be **deprecated** by changing the `scenario.status` field to `deprecated`.
```
Comment (cksnp) [inline docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md:348]: Perhaps worth adding a sentence to explain that this is essentially documentation, so deprecating a scenario is a flag to tell consumers not to rely on this scenario anymore.

**Review by cksnp — CHANGES_REQUESTED** [review]
A few small things only!

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 10]
```diff
@@ -0,0 +1,367 @@
...
+- `dataStructureId`: Filters the tracking scenarios associated with a particular data structure.
+- `dataStructureVersion`: Filters the tracking scenarios associated with a specific data structure version when used with `dataStructureId`.
+- `withHistory`: Returns a list with the history of each tracking scenario in the `includes` field of the response. The relation between tracking scenarios in `data` and history in `includes` is `id = scenarioId`.
+- `status`: Filters the tracking scenarios that match the specified status.
+
+### Compatibility Checks
+
+This request may return warnings within the `errors` field of the response:
+
+- **SchemaIncompatible**: The tracking scenario is not compatible with a specific Iglu schema version. The `type` is always `Warning`.
```
[the remark this answers is not in the document]
Comment (elyerry) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md]: I was thinking the same... I think for documentation we should probably stick with "Data Structure".

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (patofp) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md]: I think that with the latest commit this is more clear now. What are your thoughts @gleb-lobov?

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 11]
```diff
@@ -0,0 +1,367 @@
...
+- **SchemaUndecidable**: It cannot determine the compatibility of the tracking scenario with a specific Iglu schema version. The `type` is always `Warning`.
```
[the remark this answers is not in the document]
Comment (patofp) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md]: I tried to be more clear about what each error and type means.

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md (hunk 2 above)
Comment (patofp) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md]: Done

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 12]
```diff
@@ -0,0 +1,367 @@
+---
+title: "🆕 Managing tracking scenarios via the API"
+date: "2023-05-09"
+sidebar_position: 93
+---
+
+With the **Tracking Scenarios API**, you can efficiently manage tracking scenarios programmatically. Whether you want to retrieve, create, edit, publish, deprecate, or delete tracking scenarios, the API provides the necessary endpoints and functionalities.
```
[the remark this answers is not in the document]
Comment (patofp) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md]: Done

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 13]
```diff
@@ -0,0 +1,367 @@
+---
+title: "🆕 Managing tracking scenarios via the API"
+date: "2023-05-09"
```
[the remark this answers is not in the document]
Comment (patofp) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md]: Removed date

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 14]
```diff
@@ -0,0 +1,367 @@
...
+    ]
+}
+```
+
+- `data`: Contains the scenario or scenarios, depending on the request.
+- `includes`: Contains additional information, such as the history of tracking scenario changes.
+- `errors`: Contains a list of errors or warnings, depending on the request.
+
+:::info
+Some endpoints perform a validation of the tracking scenario `event`.
```
[the remark this answers is not in the document]
Comment (patofp) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md]: Removed the info box

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 15]
```diff
@@ -0,0 +1,367 @@
...
+        },
+        "additionalProperties": false
+    }
+}
+...
+```
+
+This validation checks the compatibility of a particular tracking scenario version with the source Iglu schema version referenced by `event.source`.
+
+Normally, a tracking scenario is always compatible with the source Iglu schema version when created. However, there could be cases where the schema evolves to a new version and is not backward compatible. In such cases, a **Warning** or **Error** (depending on the endpoint) can be returned in the errors field, providing information about the severity, tracking scenario ID, Iglu schema, version, and reason for the incompatibility.
```
[the remark this answers is not in the document]
Comment (patofp) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md]: This text has been changed

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md [hunk 16]
```diff
@@ -0,0 +1,367 @@
...
+        }
+    },
+    "message": "initial draft"
+}
+```
+
+- `message`: An optional field to provide a message.
+- `scenario`: The definition of the tracking scenario, which should have a defined structure. See [Validations](#validations).
+
+By default, the tracking scenario will be created with `scenario.status` set to **draft** and `scenario.version` set to `0` if not provided. These values can be changed and managed after creation. Here is an example response:
```
[the remark this answers is not in the document]
Comment (patofp) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md]: Done

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md (hunk 4 above)
Comment (patofp) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md]: Replaced in favor of "discarded".

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (gleb-lobov) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md]: Looks good 👍

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md (hunk 5 above)
Comment (patofp) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md]: Done

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md (hunk 7 above)
Comment (patofp) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md]: Hmm not sure what you mean Costas, can you elaborate a little bit on this? Thank you 😅

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md (hunk 8 above)
Comment (patofp) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md]: Agreed, good point. I added a reference to the paper.

File: docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md (hunk 9 above)
Comment (patofp) [reply docs/understanding-tracking-design/managing-tracking-scenarios-api/index.md]: I elaborated a little bit more what is the intention of 'deprecated' status

**Comment by patofp** [comment]
Continuing in PR #501

**Pull request closed without merging**